### PR TITLE
added bytecode definition

### PR DESCRIPTION
**Pull request by mportman12**
Added definition for bytecode to Glossary per [issue 242](https://github.com/redhat-documentation/supplementary-style-guide/issues/242).

In addition to peer review, this entry needs SME review. @jseseCCS can you recommend an SME reviewer who can confirm that the definition is accurate for Red Hat purposes?

### CODE REVIEW

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc [hunk 1]
```diff
@@ -520,3 +520,14 @@ Do not use "BIOS" as a generic term to refer to computer firmware. Use "firmware
 *Incorrect forms*:
 
 *See also*:
+
+[discrete]
+[[bytecode]]
+==== image:images/yes.png[yes] bytecode (noun)
+*Description*: A _bytecode_ is a non-human-readable instruction set generated by a compiler. Bytecode is typically either executed by a virtual machine (VM) or compiled again into machine code.
```
Comment (dfitzmau) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:527]: xompiled again > re-compiled ?

Comment (mportman12) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:527]: @dfitzmau Although 'recompile' is a term, it's not frequently used and is what I consider to be technical and archaic. While this glossary entry is a technical definition, I'd prefer not to use 'recompile' if there's a way to write around it.

Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:527]: Hi @mportman12 . NP. I guess it depends on what product one works on. In the OpenJDK "re-compile" is a common action when building Java applications·

Comment (bredamc) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:527]: The term "recompile" sounds more natural to me, but maybe I'm archaic too :rofl: 
I agree that it's a technical term, but it's used in a description of a technical term so it's appropriate?

Note that if we do use this term, the spelling should be "recompile" (without the hyphen), as specified in the Hyphens topic and in Merriam-Webster.

Comment (mportman12) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:527]: Ok, if 'recompile' is not archaic in this context, I'll update the text for it.

**Comment by mportman12** [comment]
Updated text for Darragh's suggestion and for SME review.

**Review by ncbaratta — DISMISSED** [review]
lgtm

**Review by dfitzmau — APPROVED** [review]

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc [hunk 2]
```diff
@@ -520,3 +520,14 @@ Do not use "BIOS" as a generic term to refer to computer firmware. Use "firmware
 *Incorrect forms*:
 
 *See also*:
+
+[discrete]
+[[bytecode]]
+==== image:images/yes.png[yes] bytecode (noun)
+*Description*: A _bytecode_ is a non-human-readable instruction set generated by a compiler. Bytecode is typically either executed by a virtual machine (VM) or recompiled into machine code. For example, Java bytecode is executed on the Java Virtual Machine (JVM).
```
Comment (bredamc) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc:527]: Suggest adding some text (or perhaps a comma) after "set" to clarify that it's an "instruction set" (noun), not an "instruction that is set" (verb)?

```suggestion
*Description*: A _bytecode_ is a non-human-readable instruction set that is generated by a compiler. Bytecode is typically either executed by a virtual machine (VM) or recompiled into machine code. For example, Java bytecode is executed on the Java Virtual Machine (JVM).
```

Comment (mportman12) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/b.adoc]: Made @bredamc's changes.

**Review by dfitzmau — APPROVED** [review]

**Review by ncbaratta — APPROVED** [review]
lgtm

**Pull request merged**